src/screens/Notifications/ConfirmEmailScreen.tsx

```diff
@@ -197,13 +238,26 @@ class ConfirmEmailScreen extends Component<Props, State> {
   };
 
   onConfirm = () => {
-    // authenticate, then:
-    return this.props.createNotificationEmail(this.props.route.params.address, {
-      onFailure: () => Alert.alert('error'),
-      onSuccess: () => {
-        this.infoContainerContent.onCodeConfirm();
+    const {
+      sessionToken,
+      notificationError,
+      storedEmail,
+      createNotificationEmail,
+      route: {
+        params: { walletsToSubscribe, email },
       },
-    });
+    } = this.props;
+    if (walletsToSubscribe?.length) {
+      if (!storedEmail && email) {
+        createNotificationEmail(email);
+      }
+      this.props.authenticate(sessionToken, this.state.code, {
+        onFailure: () => Alert.alert('error', notificationError),
```

Comment:
I don't think we have an alert on error somewhere in the designs

Reply:
Yes this case should be discussed because there are may exception to handle, but designs don't cover all of them

Reply:
please add a note here then, and please add a subtask in Jira to not forget about it

Reply:
Already done https://jira.minebest.com/browse/BTCV2-1384